webui/react/src/components/CheckpointModal.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { act, render } from '@testing-library/react';
+import { useModal } from 'hew/Modal';
+import UIProvider, { DefaultTheme } from 'hew/Theme';
+import { ConfirmationProvider } from 'hew/useConfirm';
+import { useLayoutEffect } from 'react';
+
+import { CheckpointState } from 'types';
+
+import CheckpointModalComponent, { Props } from './CheckpointModal';
+import { ThemeProvider } from './ThemeProvider';
+
+const mockUseFeature = vi.hoisted(() => vi.fn());
+vi.mock('hooks/useFeature', () => {
+  return {
+    default: () => ({
+      isOn: mockUseFeature,
+    }),
+  };
+});
+
+const mockConfig = {
```

Comment:
nit: can all of these mock config, checkpoint, experiment, etc objects be moved to `generateTestData.ts` or somewhere similar?